### DISCUSSION TITLE
Template Tags inside GrapesJS canvas

### DISCUSSION THREAD

**Discussion by daniel23ad-pulse** [GrapesJS/grapesjs#5277] [Q&A]
I am working on a project that requires both AlpineJS and GrapesJS.
One thing I have been noticing though is that when trying to implement cycles with AlpineJS's x-for (which are required to be inside a template HTML tag) inside GrapesJS's canvas for any custom Block it does not work and only has '#document-fragment' inside of it. 
I have already tried to do this according to another discussion I saw here on GitHub but this way I have problems with AlpineJS instead, as the template tag gets turned into a div tag which is not compatible with the x-for AlpineJS directive.

![Screenshot 2023-08-04 at 10 05 23](https://github.com/GrapesJS/grapesjs/assets/141316746/468dce2f-fd57-4cf4-9889-4c462c28a86b)

Any way that this could be solved?

Thanks in advance!

**Comment by yoheiyakmo** [comment]
Hi --- did you manage to find a solution / hint? I'm having a same problem regarding <template> </template> inside canvas...

**Reply by daniel23ad-pulse** [reply]
Hi! This project I was working on was left on hold for quite some time so maybe this is not the most accurate answer... but what I think I did was to render that element I needed on the server side (in my case, using PHP). To this problem in particular I did not find any direct solution though, I am sorry...
